### DISCUSSION TITLE
Allow upgrading assets (replacing conflicting files) from AssetLib

### DISCUSSION THREAD

**Discussion by Yagich** [godotengine/godot-proposals#8180] [Editor]
When downloading an asset from the AssetLib in the editor, I thought it would be a common workflow to upgrade the asset by redownloading it and installing again. Instead, the dialog explicitly disallows overwriting conflicting files. I think it should be allowed, maybe with a checkbox, to overwrite the "conflicting" files when downloading an asset.

**Comment by hsandt** [comment]
Thanks for opening this, I swore I had already opened an issue as it bothered me multiple times, but I can't find it anywhere. I will open a full proposal showing some mockup interface.

One limitation, also observed in Unity, of purely allowing file overwrites, is that it doesn't take the case of files being *removed* in future asset versions. That said, being able to overwrite existing files would already cover 80% of the cases, where asset upgrades consist in modifying and *adding* new files only. But an extra option to remove all files not present in new version would be even better.

The current workaround is to delete the old add-on folder manually (both to avoid keeping old files removed in new version, and to avoid having existing files blocking file overwrite). However, this will break many dependencies, until you install the new asset version - and even then some dependencies are not properly reconnected at that point. There is no command line to update all packages while Godot editor is closed (except 3rd party like the npm solutions mentioned in https://github.com/godotengine/godot-proposals/issues/8451#issuecomment-1816086534), so you need to keep Godot open to do this and so you will see some dependency break live... (the best you can do is to close all the scenes, but it's not perfect either).

So a weird twist is that, after the upgrade, if you got broken dependencies remaining, you close Godot editor, open your VCS and revert any changes on files with bad dependency resaved, only keeping the actual asset upgrade change. Also delete .godot folder to be safe, and finally reopen the project. In simple cases like updating a non-critical asset like GUT, it's ok, and if you're lucky, but for add-ons closely connected to your project like some game menu template you may have many more errors after old files removal.

**Reply by hsandt** [reply]
Opened https://github.com/godotengine/godot-proposals/issues/10039
